### PR TITLE
Generate addon manifest.patch

### PR DESCRIPTION
**Pull request by wsnipex**
This moves patching the addon manifest from a manually maintained patch file to autogeneration.
Since I couldn't find a way to do it all from within flatpak builder, it's a 2 step process:
1. the Makefile produces a list of all addons we ship
2. copy and run a small helper script to the kodi source to patch addon-manifest.xml

### CODE REVIEW

**Comment by wsnipex** [comment]
argh, flathub doesn't use the Makefile :(

**Comment by wsnipex** [comment]
updated, to not require the Makefile to be used.
Drawback: we now need to have addon-list.txt versioned in git, but it will be updated when running make update-addons

File: tv.kodi.Kodi.yml [hunk 1]
```diff
@@ -115,11 +115,19 @@ modules:
...
-        commit: 918ff46a659fc12502e42cd7db62eaf250e07493
+        commit: e47331c6308b7f7f9760c8813c4f70fc150d4df1
       - type: patch
         path: kodi.sh.in.patch
-      - type: patch
-        path: addon-manifest.xml.patch
+      - type: file
+        path: addon-list.txt
+        dest: build/download/
+        dest-filename: addon-list.txt
```
Comment (razzeee) [inline tv.kodi.Kodi.yml:124]: I think this can be omited, if the name matches? https://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html

Comment (wsnipex) [reply tv.kodi.Kodi.yml:124]: probably, but is it worth triggering a full rebuild?

Comment (razzeee) [reply tv.kodi.Kodi.yml:124]: Nope, but we could push it and cancel the build or include it in something that we're pushing later.

Comment (wsnipex) [reply tv.kodi.Kodi.yml:124]: sure. Let's update it with the next change.

**Pull request merged**